examples/default.yaml

```diff
@@ -166,6 +166,12 @@ caCerts:
   #   YOUR-ORGS-TRUSTED-CA-CERT-HERE
   #   -----END CERTIFICATE-----
 
+package:
+  # Upgrade the instance on first boot
+  # Reboot after upgrade if required
+  # 🟢 Builtin default: false
+  upgrade: null
+
```

Comment:
To avoid inflating the numbers of the limayaml fields, can we consider to support templating cloud-init yaml in limayaml?

Reply:
I don't want the user to have to learn cloud-init, since they sort of use lima.yaml to **not** have to do that

Reply:
The `cloud-init` support is a good feature for advanced users, just not required for _this_ particular feature

Reply:
I agree, I'm in favour of approving this PR; @AkihiroSuda are you going to veto it?

Reply:
> The user just wants up-to-date packages.

You can already do this with a dependency provisioning script (untested):

```yaml
provision:
- mode: dependency
  script: |
    apt-get update
    apt-get upgrade
    if [ -f /var/run/reboot-required ]; then
        reboot
    fi
```

This script is executed as part of `30-install-packages.sh`, just before additional packages are installed.

Reply:
I was naively hoping this to be a oneliner, both in the lima.yaml and in the cloud-init.

Reply:
> I agree, I'm in favour of approving this PR; @AkihiroSuda are you going to veto it?

@AkihiroSuda can you please let us know if you are opposed to making this change, and want to wait until we have some generic `cloudInit` mechanism? Because then we can stop arguing about this PR if it has no chance of getting merged. 😃

I think this PR provides reasonable functionality to expose via Lima directly without requiring the user to learn about `cloud-init` config files. We also simplify this to a single setting.

Also, contrary to my earlier comments, I don't think we should provide the `packages` field. So the `upgrade_packages` would be the only key I would expect to be added to `lima.yaml`.